torch/fx/symbolic_trace.py

```diff
@@ -295,6 +299,7 @@ def trace(self, root: Union[torch.nn.Module, Callable], concrete_args: Optional[
         if isinstance(root, torch.nn.Module):
             self.root = root
             fn = type(root).forward
+            self.submodule_paths = {mod: name for name, mod in root.named_modules()}
```

Comment:
Maybe also null this out when we exit `trace()`? Not sure if it can happen but e.g. if someone uses the same `Tracer` instance to first trace a module but then trace a free function, the free function trace would happen with a `submodule_paths` instance that doesn't belong to it